src/lib/utils/fetch-cost-of-carry.ts

```diff
@@ -40,9 +51,31 @@ export async function fetchCostOfCarry(
       return
     }
 
-    setCostOfCarry(
-      Number(borrowedAsset.supplyAPY) - Number(borrowedAsset.variableBorrowAPY),
+    const { collateralAmount, debtAmount } = await publicClient.readContract({
+      address: FlashMintLeveragedExtendedAddress,
+      abi: FlashMintLeveragedExtendedAbi,
+      functionName: 'getLeveragedTokenData',
+      args: [
+        inputOutputToken.arbitrumAddress as Address,
+        inputTokenAmount,
+        isMinting,
+      ],
+    })
+
+    const navProvider = new NavProvider()
+    const nav = await navProvider.getNavPrice(
+      inputOutputToken.symbol,
+      ARBITRUM.chainId,
     )
+
+    const supplyAPY = Number(borrowedAsset.supplyAPY)
+    const borrowAPY = Number(borrowedAsset.variableBorrowAPY)
+
+    const costOfCarry =
+      (Number(formatUnits(collateralAmount, 18)) * supplyAPY -
+        Number(formatUnits(debtAmount, 18)) * borrowAPY) /
```

Comment:
Make sure to check the `decimals` depending on the positions (included in leveraged token data as well). Since ETH2X or ETH3X for example have USDC included decimals would be `6` (instead of 18).